packages/@aws-cdk/integ-runner/lib/runner/integration-tests.ts

```diff
@@ -248,15 +296,33 @@ export class IntegrationTests {
   }
 
   private async discover(options: IntegrationTestsDiscoveryOptions): Promise<IntegTest[]> {
-    const patterns = options.testRegex ?? ['^integ\\..*\\.js$'];
+    const languagePresets = options.language ?? Array.from(IntegrationTests.defaultDiscoveryRegexes.keys());
+    const patterns = options.testRegex?.map((pattern) => new RegExp(pattern))
+        ?? Array.from(IntegrationTests.defaultDiscoveryRegexes.entries()).filter(
+          ([language]) => languagePresets.includes(language),
+        ).map(([_, regex]) => regex);
 
     const files = await this.readTree();
-    const integs = files.filter(fileName => patterns.some((p) => {
-      const regex = new RegExp(p);
+    const integs = files.filter(fileName => patterns.some((regex) => {
       return regex.test(fileName) || regex.test(path.basename(fileName));
     }));
 
```

Comment:
Okay so here's an idea how we could streamline the language presets a bit. Instead of `IntegrationTestsDiscoveryOptions` having all three: `testRegex` `app` and `language` we introduce a new map:
```ts
{
  "app command": ["test regex"],
  // e.g.
  "node {filePath}": ['^integ\\..*\\.js$']
}
```
(In case of `--app` and/or `--test-regex` we just amend the map accordingly)

That way you could mostly keep the current code, but add an outer loop around it:
```ts
for (const [appCommand, patterns] of Object.entries(options.theNewMapThingy)) {
  // do the old thing here
  // but also map the matched files to IntegTests
  const integs = files
    .filter(fileName => patterns.some((pattern) => {
      const regex = new RegExp(pattern);
      return regex.test(fileName) || regex.test(path.basename(fileName));
    }))
    .map(fileName => new IntegTest({
      discoveryRoot: this.directory,
      fileName,
      appCommand, // this is now the `appCommand` from `theNewMapThingy`
    })); 
}
```

What do you think?

This would probably mean we need to move the conversion from list of language presets and/or `--app`/`--test-regex` to somewhere and call it from `cli.ts`.